dds_web/__init__.py

```diff
@@ -368,9 +369,31 @@ def create_new_unit(
     days_in_available,
     days_in_expired,
 ):
-    """Create a new unit."""
+    """Create a new unit.
+
+    Rules for bucket names, which are affected by the public_id at the moment:
+    https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html
+    """
     from dds_web.database import models
 
+    error_message = ""
+    if len(public_id) > 50:
+        error_message = "The 'public_id' can be a maximum of 50 characters"
+    elif re.findall(r"[^a-zA-Z0-9.-]", public_id):
+        error_message = (
+            "The 'public_id' can only contain letters, numbers, dots (.) and hyphens (-)."
+        )
+    elif public_id[0] in [".", "-"]:
+        error_message = "The 'public_id' must begin with a letter or number."
+    elif public_id.count(".") > 2:
+        error_message = "The 'public_id' should not contain more than two dots."
+    elif public_id[:4] == "xn--":
```

Comment:
public_id.startswith() should also work. I don't think it really matters as it's such a minor copy operation.